main.js

```diff
@@ -3,21 +3,29 @@ var appState = {
   currentCharacter: 0,
 }
 
+var $header = document.createElement('h1')
+$header.textContent = 'Typing Tutor'
 var string = 'grumpy wizards make toxic brew for the evil queen and jack'
 
 for (var i = 0; i < string.length; i++) {
-  let obj = {}
-  var char = 'char'
-  obj[char] = string[i]
+  var obj = {
+    correct: true,
+    char: string[i],
+    failure: 0,
+  }
   appState.phrase.push(obj)
 }
 
 function grabSingleChar(object, index) {
+  let currentCharIndex = appState.phrase[appState.currentCharacter]
   $character = document.createElement('a')
-  $character.textContent = object[char]
+  $character.textContent = object['char']
```

Comment:
It's better to use dot notation when you know the property name. Also, `object` is kind of a crummy parameter name. See if you can think of something more descriptive.